public/app/features/dashboard-scene/sharing/ExportButton/utils.ts

```diff
@@ -0,0 +1,162 @@
+}
+
+/**
+ * Constants for grid layout calculations
+ */
+export const GRID_CELL_HEIGHT = 30; // Height of a single grid cell in pixels
+export const GRID_CELL_MARGIN = 8; // Margin between grid cells in pixels
+export const EXTRA_PADDING = 80; // Additional padding for the dashboard image
+export const MIN_DASHBOARD_HEIGHT = 400; // Minimum height of the dashboard image
+
+/**
+ * Supported image formats for export
+ */
+export type ImageFormat = 'png' | 'jpg';
+
+/**
+ * Options for generating a dashboard image
+ */
+export interface ImageGenerationOptions {
+  dashboard: DashboardScene;
+  format?: ImageFormat;
+  scale?: number;
+}
+
+/**
+ * Result of image generation attempt
+ */
+export interface ImageGenerationResult {
+  blob: Blob;
+  error?: string;
+  errorCode?: ImageGenerationError;
+}
+
+/**
+ * Calculates the total dimensions needed for the dashboard
+ * @param dashboard The dashboard scene
+ * @returns The calculated dimensions in pixels
+ */
+export function calculateDashboardDimensions(dashboard: DashboardScene): { width: number; height: number } {
+  // Get all panels from the layout manager
+  const panels = dashboard.state.body.getVizPanels();
+
+  // Default dimensions
+  let maxWidth = window.innerWidth;
+  let maxHeight = MIN_DASHBOARD_HEIGHT;
+
+  // Calculate dimensions based on grid positions
+  for (const panel of panels) {
+    const parent = panel.parent;
+    if (parent instanceof DashboardGridItem) {
+      const gridPos = parent.state;
+      const y = gridPos.y ?? 0;
+      const height = gridPos.height ?? 0;
+      const x = gridPos.x ?? 0;
+      const width = gridPos.width ?? 0;
+
+      // Calculate total height including margins between cells
+      const panelBottom =
+        y * (GRID_CELL_HEIGHT + GRID_CELL_MARGIN) + height * GRID_CELL_HEIGHT + (height - 1) * GRID_CELL_MARGIN;
+      // Calculate total width including margins between cells
+      const panelRight =
+        x * (GRID_CELL_HEIGHT + GRID_CELL_MARGIN) + width * GRID_CELL_HEIGHT + (width - 1) * GRID_CELL_MARGIN;
+
+      maxHeight = Math.max(maxHeight, panelBottom);
+      maxWidth = Math.max(maxWidth, panelRight);
+    }
+  }
+
+  // Add extra padding for better visualization
+  return {
+    width: maxWidth + EXTRA_PADDING,
+    height: maxHeight + EXTRA_PADDING,
+  };
+}
+
+/**
+ * Generates a dashboard image using the renderer service
+ * @param options The options for image generation
+ * @returns A promise that resolves to the image generation result
+ */
+export async function generateDashboardImage({
+  dashboard,
+  format = 'png',
+  scale = config.rendererDefaultImageScale || 1,
+}: ImageGenerationOptions): Promise<ImageGenerationResult> {
+  if (!config.rendererAvailable) {
+    return {
+      blob: new Blob(),
+      error: 'Image renderer plugin not installed',
+      errorCode: ImageGenerationError.RENDERER_NOT_AVAILABLE,
+    };
+  }
+
+  try {
+    const dimensions = calculateDashboardDimensions(dashboard);
```

Comment:
I don't think this is really needed. Sending `height=-1` will make the image renderer scroll through the dashboard and take the appropriate height. The width could be configured in the UI (as in the panel image sharing workflow). 
Unless you found an issue with sending `height=-1` but we should fix it then because it's what we used to generate the dashboard image in the reporting feature.